src/fwrite.c

```diff
@@ -836,7 +827,11 @@ void fwriteMain(fwriteMainArgs args)
            zbuffSize / MEGA, nth, errno, strerror(errno));
       // # nocov end
     }
+  } else {
+    // if no is_gzip, malloc 0 for allowing freeing zbuffPool
+    zbuffPool = malloc(0);
```

Comment:
The C standard (even C89) allows giving null pointers to `free` (and allows `malloc(0)` to return a null pointer). Circa-2016 (R-3.3.0 release year) C standard libraries should all handle `free(NULL)` without a problem.

Initialising `char * zbuffPool = NULL` on line 801 above and skipping the `else` branch here would have been sufficient.